File: Angular_Frontend/skiresort/src/app/classes/pass-request.ts
import {formatDate} from "@angular/common";

export class PassRequest {
  unitPrice: number;
  firstName: string;
  lastName: string ;
  startDate: any;
  endDate: any;
  birthDate: string;
  usesTotal: number;


  constructor(unitPrice: number, firstName: string, lastName: string, startDate: Date, endDate: Date, birthDate: Date, usesTotal: number) {
    this.unitPrice = unitPrice;
    this.firstName = firstName;
    this.lastName = lastName;
    if(startDate != null && endDate != null){
      this.startDate = formatDate(startDate,'yyyy-MM-dd\'T\'HH:mm:ss.SSS', 'en-US');
      this.endDate = formatDate(endDate,'yyyy-MM-dd\'T\'HH:mm:ss.SSS', 'en-US');
    }
    else{
        this.startDate = null;
        this.endDate = null;
    }
    this.birthDate = formatDate(birthDate,'yyyy-MM-dd', 'en-US');
    this.usesTotal = usesTotal;
  }
}

File: Spring_Backend/src/main/java/polsl/tab/skiresort/api/passes/request/PassRequest.java
package polsl.tab.skiresort.api.passes.request;

import java.sql.Date;
import java.sql.Timestamp;

public class PassRequest {

    private Float unitPrice;

    private Timestamp startDate; // If quantity pass - send null

    private Timestamp endDate;   // If quantity pass - send null

    private String firstName;

    private String lastName;

    private Date birthDate;

    private Integer usesTotal;  // If time pass - send null

    public Float getUnitPrice() {
        return unitPrice;
    }

    public void setUnitPrice(Float unitPrice) {
        this.unitPrice = unitPrice;
    }

    public Timestamp getStartDate() {
        return startDate;
    }

    public void setStartDate(Timestamp startDate) {
        this.startDate = startDate;
    }

    public Timestamp getEndDate() {
        return endDate;
    }

    public void setEndDate(Timestamp endDate) {
        this.endDate = endDate;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Date getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public Integer getUsesTotal() {
        return usesTotal;
    }

    public void setUsesTotal(Integer usesTotal) {
        this.usesTotal = usesTotal;
    }
}
